src/main/java/com/google/sps/servlets/ChatServlet.java

```diff
@@ -0,0 +1,118 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity; 
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Streams;
+import com.google.gson.Gson;
+import com.google.template.soy.SoyFileSet;
+import com.google.template.soy.tofu.SoyTofu;
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+
+@WebServlet("/chat")
+public class ChatServlet extends HttpServlet{
+    private static final String MESSAGE_TEXT_PROPERTY = "message-text";
+    private static final String TIMESTAMP_PROPERTY = "timestamp";
+    private static final String MESSAGE_KIND = "Message";
+
+  /**
+   * When called, doGet will query all previously posted messages to the chat,
+   * and render the messages into a HTML template.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    // Calls query on all entities of type Message
+    Query messageQuery = new Query(MESSAGE_KIND).addSort(TIMESTAMP_PROPERTY, SortDirection.ASCENDING);
+    PreparedQuery preparedMessageQuery = datastore.prepare(messageQuery);
+
+    // Creates list of the messages text
+    ImmutableList<String> messagesList = Streams.stream(preparedMessageQuery.asIterable()).map(message -> 
+        (String) message.getProperty(MESSAGE_TEXT_PROPERTY)).collect(toImmutableList());
+
+    // Data will be passed in as a list of messages in a map (needed for template)
+    ImmutableMap<String, ImmutableList<String>> data = ImmutableMap.of("messages", messagesList);
```

Comment:
could create a helper function along the lines of getMessagesMap() and move all the code in this line and above to the helper function